ticket_upgrade_prediction/pipeline.py

```diff
@@ -340,6 +333,46 @@ def concat_df_with_oh_encoding(self) -> None:
             columns=self.get_oh_cols()
         )
 
+    def scale_final_dataset(self) -> None:
+        self.concat_df_with_oh_encoding()
+        X_train, X_test, y_train, y_test = train_test_split(
+            self.df.drop(columns=self.target),
+            self.df[self.target],
+            test_size=0.2,
+            random_state=69420,
+            stratify=self.df[self.target]
+            if self.model_type == "predict_upgrade"
+            else None,
+        )
+        X_train[self.get_cols_to_scale()] = self.scaler.fit_transform(
+            X_train[self.get_cols_to_scale()]
+        )
+        X_test = self.scaler.transform(X_test[self.get_cols_to_scale()])
+        self.df = {
+            "X_train": X_train,
+            "X_test": X_test,
+            "y_train": y_train,
+            "y_test": y_test,
+        }
```

Comment:
name df is misleading here, since it is a `dict[str, pd.DataFrame]`, not pd.DataFrame.
What about using dataclass for it?
For example:

```
@dataclass
class Datasets:
    X_train: pd.DataFrame
    X_test: pd.DataFrame
    y_train: pd.Series #or np.ndarray, im not sure
    y_test: pd.Series #or np.ndarray, im not sure
```

Reply:
And this function could return an object of this dataclass here.
We could also make `.merge_files()` method return df. 
So that while performing kfold hyperopt, we could retrive not splitted dataframe using:
```
pipeline = Pipeline()
df = pipeline.merge_files()
```
And if we wanted to train final model on params found in kfold cross validation, we could do: 
```
pipeline = Pipeline(...)
_ = pipeline.merge_files()
datasets = pipeline.scale_final_dataset()
```

Reply:
That would require changes in line 26 aswell

Reply:
on one hand yes, but feature scaling will be problematic then. you could merge it inside i.e. jupyter notebook for cross-validation imo

Reply:
Why scaling would be problematic?